pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -96,7 +112,21 @@ func (a *Actuator) Exists(ctx context.Context, machine *machinev1.Machine) (bool
 	// When create()/update() try to store machineSpec/status this might result in
 	// "Operation cannot be fulfilled; the object has been modified; please apply your changes to the latest version and try again."
 	// Therefore we don't close the scope here and we only store spec/status atomically either in create()/update()"
-	return newReconciler(scope).exists()
+	_, exists, err := newReconciler(scope).exists()
+	if isInvalidMachineConfigurationError(err) {
```

Comment:
The main machine controller should try and recognise this situation no? Does it not handle invalid configuration on exists? 

What would happen if there is an actual Machine there, the exists call returns false, but with an error, so we requeue? We only set the phase to failed when it's empty so this should be ok I guess

Reply:
1. I would think so as well, because of the comments that are spread in these sections of code, but unfortunately it doesn't recognize the situation.

2. With this fix, we distinguish whether we get a 404 error or just a regular error and on that basis, we determine what we want to do next. Either re-queue for a regular error, or if the machine doesn't exist (e.g. due to the configuration not being valid), we want to delete/fail that machine.